mlog/log.go

```diff
@@ -297,13 +300,27 @@ func (l *Logger) RemoveTargets(ctx context.Context, f func(ti TargetInfo) bool)
 	fc := func(tic logr.TargetInfo) bool {
 		return f(TargetInfo(tic))
 	}
-	return l.logrLogger.Logr().RemoveTargets(ctx, fc)
+	return l.getLogger().Logr().RemoveTargets(ctx, fc)
 }
 
 // EnableMetrics enables metrics collection by supplying a MetricsCollector.
 // The MetricsCollector provides counters and gauges that are updated by log targets.
 func (l *Logger) EnableMetrics(collector logr.MetricsCollector) error {
-	return l.logrLogger.Logr().SetMetricsCollector(collector)
+	return l.getLogger().Logr().SetMetricsCollector(collector)
+}
+
+// getLogger is a concurrent safe getter of the logr logger
+func (l *Logger) getLogger() *logr.Logger {
+	defer l.mutex.RUnlock()
```

Comment:
Wondering if we should also use the atomic approach we used in the earlier PR? 0/5 though.

Reply:
i do not see why not, also 0/5 here. I guess mutexes are slower and for a logger an atomic.Value will me more performant. I do not know how well atomic. Value plays with copies though

Reply:
Since from the docs it says `Once Store has been called, a Value must not be copied.` ref https://golang.org/pkg/sync/atomic/#Value

Reply:
Yeah, doesn't make sense if there are copies involved. :+1: